### PR TITLE
Add a no-op ViewManager (closes #614).

### LINKED ISSUE #614: Decide whether to add  a no-op ViewManager.
Currently, `Stats.getViewManager()` returns null if no implementation is available.  We could instead return a no-op ViewManager, for consistency with `Stats.getStatsRecorder()` and most of the tracing classes.  However, a no-op ViewManager is much less useful than a no-op StatsRecorder, because the ViewManager isn't used by instrumented code.  The ViewManager is used by code that may already need to check whether stats collection is enabled.  That code would need to call an `isNoop()` method instead of checking for null.

I looked at the other packages quickly, and the only other classes that use null instead of no-op implementations are `RunningSpanStore` and `SampledSpanStore`.

EDIT: [ViewManager](https://github.com/census-instrumentation/opencensus-java/blob/master/core/src/main/java/io/opencensus/stats/ViewManager.java) contains the methods for registering views and querying the stats collected for those views.

### PR DESCRIPTION
**Pull request by sebright**
NoopViewManager maintains a map of views so that it can return empty ViewDatas
from getView.  It also prevents different views with the same name from being
registered.

### CODE REVIEW

File: core/src/main/java/io/opencensus/stats/NoopStats.java [hunk 1]
```diff
@@ -72,4 +93,50 @@ public void record(TagContext tags, MeasureMap measureValues) {
...
+        if (existing == null) {
+          throw new IllegalArgumentException("View is not registered.");
+        } else {
+          return ViewData.create(
+              existing,
+              Collections.<List<TagValue>, AggregationData>emptyMap(),
+              existing
+                  .getWindow()
+                  .match(
+                      // TODO(sebright): Should we use the zero timestamp or the current time?
```
Comment (songy23) [inline core/src/main/java/io/opencensus/stats/NoopStats.java]: I think zero timestamp is better.

Comment (sebright) [reply core/src/main/java/io/opencensus/stats/NoopStats.java]: Thanks.  I removed the TODO.

**Comment by sebright** [comment]
@bogdandrutu I created an issue: #679

**Comment by sebright** [comment]
Thanks for the review!

**Pull request merged**